app/Traits/Controller/v1/HasSemesters.php

```diff
@@ -17,17 +18,23 @@ trait HasSemesters
     /**
      * Retrieve the specified user.
      *
-     * @param  string $semester_id
+     * @param  string  $semester_id
+     * @param  boolean $hideData
      * @return Semester
      */
-    protected function getSemester(string $semester_id=null)
+    protected function getSemester(string $semester_id=null, bool $hideData=false)
```

Comment:
This is specific only for your controller. Do not change this trait pls bcs it causes a bug, explained just under: